core/conversion/converters/impl/cast.cpp

```diff
@@ -0,0 +1,82 @@
+#include <torch/torch.h>
+#include "core/conversion/converters/converter_util.h"
+#include "core/conversion/converters/converters.h"
+#include "core/util/prelude.h"
+#include "core/util/trt_util.h"
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace impl {
+namespace {
+
+auto cast_registrations TRTORCH_UNUSED =
+    RegisterNodeConversionPatterns()
+        .pattern(
+            {"aten::to.dtype(Tensor self, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor)",
+             [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+               auto self = args[0].ITensorOrFreeze(ctx);
+               auto output_dtype = args[1].unwrapToScalar().to<int64_t>();
+               auto aten_to_trt_dtype_map = util::get_aten_trt_type_map();

```

Comment:
We can just use ScalarToTRTDataType. Use the opt version if you want to handle the error yourself

Reply:
done